cmd/commands/init.go

```diff
@@ -0,0 +1,122 @@
+package commands
+
+import (
+	"io/ioutil"
+	"path"
+
+	"github.com/urfave/cli"
+
+	cmn "github.com/tendermint/go-common"
+)
+
+var InitCmd = cli.Command{
+	Name:      "init",
+	Usage:     "Initialize a basecoin blockchain",
+	ArgsUsage: "",
+	Action: func(c *cli.Context) error {
+		return cmdInit(c)
+	},
+	Flags: []cli.Flag{
+		ChainIDFlag,
+	},
+}
+
+func cmdInit(c *cli.Context) error {
+	rootDir := BasecoinRoot("")
+
+	cmn.EnsureDir(rootDir, 0777)
+
+	// initalize basecoin
+	genesisFile := path.Join(rootDir, "genesis.json")
+	privValFile := path.Join(rootDir, "priv_validator.json")
+	key1File := path.Join(rootDir, "key.json")
+	key2File := path.Join(rootDir, "key2.json")
+
+	if err := ioutil.WriteFile(genesisFile, []byte(genesisJSON), 0644); err != nil {
```

Comment:
This is one of those things that looks like a good idea at first and is discovered to have been a mistake later. If we ever want this functionality to be useable from a library, its a pain to have to go pull out the exits - best to just return errors.